api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto

```diff
@@ -450,6 +452,34 @@ message HttpConnectionManager {
   // with `prefix` match set to `/dir`. Defaults to `false`. Note that slash merging is not part of
   // `HTTP spec <https://tools.ietf.org/html/rfc3986>` and is provided for convenience.
   bool merge_slashes = 33;
+
+  // Configuration of local reply send by envoy for http responses. Allows to define custom rules
+  // which maps response attribute returned by envoy to different one. Currently mapping of
+  // status code is implemented.
+  message SendLocalReplyConfig {
+    // Configuration which allows to define custom rules for matching local response.
+    // It works like and operator e.g: if you define status code 404 and body pattern
+    // then response must match both
+    message Match {
```

Comment:
I really, really, really would like to unify our HTTP match capabilities, but maybe we defer this to v4 as I think any other solution will be too disruptive. @htuch any thoughts here?

Reply:
I think that we should tackle this in v4, but meanwhile, for new mathers, can we align them behind either access log or TAP matchers as the basis (even if only subsetted)? I'd prefer not to add yet another matcher format (and surrounding code cruft for parsing and executing match expressions).

Reply:
My high level question here is do we really need a match/rewrite semantics here? Since local reply is the output from Envoy (either from core HCM, or extensions), can we change the code of local reply to emit some structured data, and map it to the actual response based on config? A `Rewrite` here seems based on how current local reply works.

Reply:
I am not sure if we want always to map some response. I thought that it would be nice to just map specifice responses to another one like it has been written in  #7537.

Reply:
IMO it's a bit easier to allow the operator to have generic rules for local replies versus having to account for every possible location where a reply is generated, but I could definitely be convinced otherwise.

Reply:
I understand the context of #7537, but I'd like this to be generic enough to make it possible to convert all local reply to JSON. Particularly can `Rewrite` be some sort of formatter instead?

`Rewrite` proposed here seems based on how current local reply works (i.e. take what current sendLocalReply send, and rewrite the response header/body), instead can we just make sendLocalReply emit what formatter says for those?

Reply:
Yeah agreed this makes sense to rethink rewrite is some combination of both rewriting fields as well as a final format phase?